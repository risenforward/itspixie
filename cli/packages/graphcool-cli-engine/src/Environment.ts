import { Output } from './index'
import { Config } from './Config'
import { Args } from './types/common'
import { Cluster } from './Cluster'
import * as fs from 'fs-extra'
import * as yaml from 'js-yaml'
import Variables from './GraphcoolDefinition/Variables'
import { InternalRC } from './types/rc'
import { ClusterNotFound } from './errors/ClusterNotFound'

export class Environment {
  out: Output
  config: Config
  args: Args
  activeCluster: Cluster
  globalRC: InternalRC = {}
  clusters: Cluster[]
  platformToken?: string
  constructor(out: Output, config: Config) {
    this.out = out
    this.config = config
  }

  async load(args: Args) {
    await this.loadGlobalRC()
  }

  clusterByName(name: string, throws: boolean = false): Cluster | undefined {
    const cluster = this.clusters.find(c => c.name === name)
    if (!throws) {
      return cluster
    }

    if (!cluster) {
      throw new ClusterNotFound(name)
    }

    return cluster
  }

  setToken(token: string | undefined) {
    this.globalRC.platformToken = token
  }

  addCluster(cluster: Cluster) {
    this.clusters.push(cluster)
  }

  saveGlobalRC() {
    const rc = {
      platformToken: this.globalRC.platformToken,
      clusters: this.getLocalClusterConfig(),
    }
    const rcString = yaml.safeDump(rc)
    fs.writeFileSync(this.config.globalRCPath, rcString)
  }

  setActiveCluster(cluster: Cluster) {
    this.activeCluster = cluster
  }

  private async loadGlobalRC(): Promise<void> {
    const globalFile =
      this.config.globalRCPath && fs.pathExistsSync(this.config.globalRCPath)
        ? fs.readFileSync(this.config.globalRCPath, 'utf-8')
        : null
    if (globalFile) {
      this.globalRC = await this.loadYaml(globalFile, this.config.globalRCPath)
      this.clusters = this.initClusters(this.globalRC)
      this.platformToken =
        this.globalRC.platformToken || process.env.GRAPHCOOL_PLATFORM_TOKEN
    }
  }

  private async loadYaml(
    file: string | null,
    filePath: string | null = null,
  ): Promise<any> {
    if (file) {
      let content
      try {
        content = yaml.safeLoad(file)
      } catch (e) {
        this.out.error(`Yaml parsing error in ${filePath}: ${e.message}`)
      }
      const variables = new Variables(
        this.out,
        filePath || 'no filepath provided',
        this.args,
      )
      content = await variables.populateJson(content)

      return content
    } else {
      return {}
    }
  }

  private getClustersFromRC(rc: InternalRC): Cluster[] {
    if (!rc.clusters) {
      return []
    }
    return Object.keys(rc.clusters).map(name => {
      const cluster = rc.clusters![name]
      return new Cluster(name, cluster.host, cluster.clusterSecret, true)
    })
  }

  private initClusters(rc: InternalRC): Cluster[] {
    const rcClusters = this.getClustersFromRC(rc)
    const sharedClusters = this.getSharedClusters(rc)
    return [...rcClusters, ...sharedClusters]
  }

  private getSharedClusters(rc: InternalRC): Cluster[] {
    return this.config.sharedClusters.map(clusterName => {
      return new Cluster(
        clusterName,
        this.config.sharedEndpoint,
        rc.platformToken!,
        false,
      )
    })
  }

  private getLocalClusterConfig() {
    return this.clusters.filter(c => c.local).reduce((acc, cluster) => {
      return {
        ...acc,
        [cluster.name]: {
          host: cluster.baseUrl,
          clusterSecret: cluster.token,
        },
      }
    }, {})
  }
}
