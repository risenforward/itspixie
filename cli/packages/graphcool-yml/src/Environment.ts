import { Args } from './types/common'
import { Cluster } from './Cluster'
import * as fs from 'fs-extra'
import * as yaml from 'js-yaml'
import { ClusterNotFound } from './errors/ClusterNotFound'
import { Variables } from './Variables'
import { IOutput, Output } from './Output'
import * as path from 'path'
import * as os from 'os'
import chalk from 'chalk'
import 'isomorphic-fetch'
import { RC } from './index'
import { DatabaseRC } from './types/rc'
import { EnvironmentMigrator } from './EnvironmentMigrator'
const debug = require('debug')('Environment')

const isDev = (process.env.ENV || '').toLowerCase() === 'dev'

export class Environment {
  sharedClusters: string[] = [
    'graphcool-eu1',
    'graphcool-us1',
    'shared-public-demo',
  ]
  clusterEndpointMap: { [key: string]: string } = {
    'graphcool-eu1': 'https://graphcool-eu1.graphcool.cloud',
    'graphcool-us1': 'https://graphcool-us1.graphcool.cloud',
    'shared-public-demo': isDev
      ? 'https://dev.database-beta.graph.cool'
      : 'https://database-beta.graph.cool',
  }
  args: Args
  activeCluster: Cluster
  globalRC: RC = {}
  databaseRC: DatabaseRC = {}
  clusters: Cluster[]
  out: IOutput
  home: string
  rcPath: string
  constructor(home: string, out: IOutput = new Output()) {
    this.out = out
    this.home = home

    this.rcPath = path.join(this.home, '.graphcoolrc')
    const migrator = new EnvironmentMigrator(home, out)
    debug(`migrating`)
    migrator.migrate()
  }

  async load(args: Args) {
    await Promise.all([this.loadGlobalRC(), this.setSharedClusters()])
  }

  async setSharedClusters() {
    // TODO: reenable

    try {
      const res = await fetch('https://stats.graph.cool/', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        } as any,
        body: JSON.stringify({
          query: `
        {
          publicClusters {
            name
            endpoint
            description
          }
        }
        `,
        }),
      })
      const json = await res.json()
      if (
        json &&
        json.data &&
        json.data.publicClusters &&
        Array.isArray(json.data.publicClusters) &&
        json.data.publicClusters.length > 0
      ) {
        this.sharedClusters = json.data.publicClusters.map(c => c.name)
        this.clusterEndpointMap = json.data.publicClusters.reduce(
          (acc, curr) => {
            return {
              ...acc,
              [curr.name]: curr.endpoint,
            }
          },
          {},
        )

        debug(this.sharedClusters)
        debug(this.clusterEndpointMap)
      }
    } catch (e) {
      debug(e)
      //
    }
  }

  clusterByName(name: string, throws: boolean = false): Cluster | undefined {
    const cluster = this.clusters.find(c => c.name === name)
    if (!throws) {
      return cluster
    }

    if (!cluster) {
      throw new ClusterNotFound(name)
    }

    return cluster
  }

  setToken(token: string | undefined) {
    this.databaseRC.cloudSessionKey = token
  }

  addCluster(cluster: Cluster) {
    const existingClusterIndex = this.clusters.findIndex(
      c => c.name === cluster.name,
    )
    if (existingClusterIndex > -1) {
      this.clusters.splice(existingClusterIndex, 1)
    }
    this.clusters.push(cluster)
  }

  saveGlobalRC() {
    const rc = {
      'graphcool-1.0': {
        cloudSessionKey: this.databaseRC.cloudSessionKey,
        clusters: this.getLocalClusterConfig(),
      },
    }
    if (this.globalRC['graphcool-framework']) {
      rc['graphcool-framework'] = this.globalRC['graphcool-framework']
    }
    // parse & stringify to rm undefined for yaml parser
    const rcString = yaml.safeDump(JSON.parse(JSON.stringify(rc)))
    fs.writeFileSync(this.rcPath, rcString)
  }

  setActiveCluster(cluster: Cluster) {
    this.activeCluster = cluster
  }

  async loadGlobalRC(): Promise<void> {
    const globalFile =
      this.rcPath && fs.pathExistsSync(this.rcPath)
        ? fs.readFileSync(this.rcPath, 'utf-8')
        : undefined
    await this.parseGlobalRC(globalFile)
  }

  async parseGlobalRC(globalFile?: string): Promise<void> {
    if (globalFile) {
      this.globalRC = await this.loadYaml(globalFile, this.rcPath)
      this.databaseRC = this.globalRC['graphcool-1.0'] || {}
    }
    this.clusters = this.initClusters(this.databaseRC)
  }

  private async loadYaml(
    file: string | null,
    filePath: string | null = null,
  ): Promise<any> {
    if (file) {
      let content
      try {
        content = yaml.safeLoad(file)
      } catch (e) {
        throw new Error(`Yaml parsing error in ${filePath}: ${e.message}`)
      }
      const variables = new Variables(
        filePath || 'no filepath provided',
        this.args,
        this.out,
      )
      content = await variables.populateJson(content)

      return content
    } else {
      return {}
    }
  }

  private initClusters(rc: DatabaseRC): Cluster[] {
    const rcClusters = this.getClustersFromRC(rc)
    const sharedClusters = this.getSharedClusters(rc)
    return [...rcClusters, ...sharedClusters]
  }

  private getSharedClusters(rc: DatabaseRC): Cluster[] {
    return this.sharedClusters.map(clusterName => {
      return new Cluster(
        clusterName,
        this.clusterEndpointMap[clusterName],
        rc && rc.cloudSessionKey,
        false,
      )
    })
  }

  private getClustersFromRC(rc: DatabaseRC): Cluster[] {
    if (!rc.clusters) {
      return []
    }
    return Object.keys(rc.clusters).map(name => {
      const cluster = rc.clusters![name]
      return new Cluster(name, cluster.host, cluster.clusterSecret, true)
    })
  }

  private getLocalClusterConfig() {
    return this.clusters.filter(c => c.local).reduce((acc, cluster) => {
      return {
        ...acc,
        [cluster.name]: {
          host: cluster.baseUrl,
          clusterSecret: cluster.clusterSecret,
        },
      }
    }, {})
  }
}
