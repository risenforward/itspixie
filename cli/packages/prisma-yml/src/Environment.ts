import { Args } from './types/common'
import { Cluster } from './Cluster'
import * as fs from 'fs-extra'
import * as yaml from 'js-yaml'
import { ClusterNotFound } from './errors/ClusterNotFound'
import { Variables } from './Variables'
import { IOutput, Output } from './Output'
import * as path from 'path'
import * as os from 'os'
import chalk from 'chalk'
import 'isomorphic-fetch'
import { RC } from './index'
import { ClusterNotSet } from './errors/ClusterNotSet'
const debug = require('debug')('Environment')

const isDev = (process.env.ENV || '').toLowerCase() === 'dev'

export class Environment {
  sharedClusters: string[] = ['prisma-eu1', 'prisma-us1']
  clusterEndpointMap: { [key: string]: string } = {
    'prisma-eu1': 'https://eu1.prisma.sh',
    'prisma-us1': 'https://us1.prisma.sh',
  }
  args: Args
  activeCluster: Cluster
  globalRC: RC = {}
  clusters: Cluster[]
  out: IOutput
  home: string
  rcPath: string
  constructor(home: string, out: IOutput = new Output()) {
    this.out = out
    this.home = home

    this.rcPath = path.join(this.home, '.prisma/config.yml')
    fs.mkdirpSync(path.dirname(this.rcPath))
  }

  async load(args: Args) {
    await Promise.all([this.loadGlobalRC(), this.setSharedClusters()])
  }

  get cloudSessionKey(): string | undefined {
    return process.env.PRISMA_CLOUD_SESSION_KEY || this.globalRC.cloudSessionKey
  }

  async setSharedClusters() {
    try {
      const res = await fetch('https://stats.graph.cool/', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        } as any,
        body: JSON.stringify({
          query: `
        {
          publicClusters {
            name
            endpoint
            description
          }
        }
        `,
        }),
      })
      const json = await res.json()
      if (
        json &&
        json.data &&
        json.data.publicClusters &&
        Array.isArray(json.data.publicClusters) &&
        json.data.publicClusters.length > 0
      ) {
        this.sharedClusters = json.data.publicClusters.map(c => c.name)
        this.clusterEndpointMap = json.data.publicClusters.reduce(
          (acc, curr) => {
            return {
              ...acc,
              [curr.name]: curr.endpoint,
            }
          },
          {},
        )

        debug(this.sharedClusters)
        debug(this.clusterEndpointMap)
      }
    } catch (e) {
      debug(e)
      //
    }
  }

  clusterByName(name: string, throws: boolean = false): Cluster | undefined {
    const cluster = this.clusters.find(c => c.name === name)
    if (!throws) {
      return cluster
    }

    if (!cluster) {
      if (!name) {
        throw new ClusterNotSet()
      }
      throw new ClusterNotFound(name)
    }

    return cluster
  }

  setToken(token: string | undefined) {
    this.globalRC.cloudSessionKey = token
  }

  addCluster(cluster: Cluster) {
    const existingClusterIndex = this.clusters.findIndex(
      c => c.name === cluster.name,
    )
    if (existingClusterIndex > -1) {
      this.clusters.splice(existingClusterIndex, 1)
    }
    this.clusters.push(cluster)
  }

  removeCluster(name: string) {
    this.clusters = this.clusters.filter(c => c.name !== name)
  }

  saveGlobalRC() {
    const rc = {
      cloudSessionKey: this.globalRC.cloudSessionKey,
      clusters: this.getLocalClusterConfig(),
    }
    // parse & stringify to rm undefined for yaml parser
    debug('saving global rc')
    debug(rc)
    const rcString = yaml.safeDump(JSON.parse(JSON.stringify(rc)))
    fs.writeFileSync(this.rcPath, rcString)
  }

  setActiveCluster(cluster: Cluster) {
    this.activeCluster = cluster
  }

  async loadGlobalRC(): Promise<void> {
    const globalFile =
      this.rcPath && fs.pathExistsSync(this.rcPath)
        ? fs.readFileSync(this.rcPath, 'utf-8')
        : undefined
    await this.parseGlobalRC(globalFile)
  }

  async parseGlobalRC(globalFile?: string): Promise<void> {
    if (globalFile) {
      this.globalRC = await this.loadYaml(globalFile, this.rcPath)
    }
    this.clusters = this.initClusters(this.globalRC)
  }

  private async loadYaml(
    file: string | null,
    filePath: string | null = null,
  ): Promise<any> {
    if (file) {
      let content
      try {
        content = yaml.safeLoad(file)
      } catch (e) {
        throw new Error(`Yaml parsing error in ${filePath}: ${e.message}`)
      }
      const variables = new Variables(
        filePath || 'no filepath provided',
        this.args,
        this.out,
      )
      content = await variables.populateJson(content)

      return content
    } else {
      return {}
    }
  }

  private initClusters(rc: RC): Cluster[] {
    const rcClusters = this.getClustersFromRC(rc)
    const sharedClusters = this.getSharedClusters(rc)
    return [...rcClusters, ...sharedClusters]
  }

  private getSharedClusters(rc: RC): Cluster[] {
    return this.sharedClusters.map(clusterName => {
      return new Cluster(
        this.out,
        clusterName,
        this.clusterEndpointMap[clusterName],
        rc && rc.cloudSessionKey,
        false,
        true,
      )
    })
  }

  private getClustersFromRC(rc: RC): Cluster[] {
    if (!rc.clusters) {
      return []
    }
    return Object.keys(rc.clusters).map(name => {
      const cluster = rc.clusters![name]
      return new Cluster(
        this.out,
        name,
        cluster.host,
        cluster.clusterSecret,
        isLocal(cluster.host),
        true,
      )
    })
  }

  private getLocalClusterConfig() {
    return this.clusters.filter(c => c.local).reduce((acc, cluster) => {
      return {
        ...acc,
        [cluster.name]: {
          host: cluster.baseUrl,
          clusterSecret: cluster.clusterSecret,
        },
      }
    }, {})
  }
}

export const isLocal = hostname =>
  hostname.includes('localhost') || hostname.includes('127.0.0.1')
