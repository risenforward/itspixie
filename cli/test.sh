#!/bin/bash

set -e

cd packages/prisma-cli-engine
npm run build
npm test
cd ../../packages/prisma-cli-core
npm test
cd ../../packages/prisma-yml
npm test
