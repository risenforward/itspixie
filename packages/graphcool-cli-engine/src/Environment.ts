import * as yaml from 'js-yaml'
import * as path from 'path'
import fs from './fs'
import { Output } from './Output/index'
import { EnvironmentConfig } from './types'
import { Client } from './Client/Client'
import EnvDoesntExistError from './errors/EnvDoesntExistError'

const envPath = path.join(process.cwd(), '.graphcoolrc')

class Environment {
  env: EnvironmentConfig
  out: Output
  client: Client

  constructor(out: Output, client: Client) {
    this.out = out
  }

  public initEmptyEnvironment() {
    this.env = {
      default: null,
      environments: {}
    }
  }

  get default(): string | null {
    if (this.env.default) {
      return this.env.environments[this.env.default]
    }

    return null
  }

  public load() {
    if (fs.existsSync(envPath)) {
      try {
        this.env = yaml.safeLoad(fs.readFileSync(envPath, 'utf-8'))
      } catch (e) {
        this.out.error(`Error in .graphcoolrc: `)
        this.out.error(e.message)
        process.exit(1)
      }
    } else {
      this.initEmptyEnvironment()
    }
  }

  public save() {
    const file = yaml.safeDump(this.env)
    fs.writeFileSync(envPath, file)
  }

  public setEnv(name: string, projectId: string) {
    this.env.environments[name] = projectId
  }

  public async set(name: string, projectId: string) {
    const version = await this.client.getProjectVersion(projectId)

    this.env.environments[name] = projectId
  }

  public setDefault(name: string) {
    if (!this.env.environments[name]) {
      throw new Error(`Environment ${name} doesn't exist in local .graphcoolrc definition`)
    }
    this.env.default = name
  }

  public rename(oldName: string, newName: string) {
    const oldEnv = this.env.environments[oldName]

    if (!oldEnv) {
      throw new Error(`Environment ${oldName} doesn't exist`)
    }

    delete this.env.environments[oldName]
    if (this.env.default === oldName) {
      this.setDefault(newName)
    }
    this.env.environments[newName] = oldEnv
  }

  public remove(envName: string) {
    const oldEnv = this.env.environments[envName]

    if (!oldEnv) {
      throw new Error(`Environment ${envName} doesn't exist`)
    }

    delete this.env.environments[envName]
  }

  public deleteIfExist(projectIds: string[]) {
    projectIds.forEach(projectId => {
      const envName = Object.keys(this.env.environments).find(envName => this.env.environments[envName] === projectId)
      if (envName) {
        delete this.env.environments[envName]
      }
    })
  }

  public async getProjectId({project, env, skipDefault}: {project?: string, env?: string, skipDefault?: boolean}): Promise<string | null> {
    if (project) {
      const projects = await this.client.fetchProjects()
      const foundProject = projects.find(p => p.id === project || p.alias === project)
      if (!foundProject) {
        throw new Error(`Project with alias or id "${project}" could not be found in this account`)
      }
      return foundProject.id
    }

    if (env) {
      const projectId = this.env.environments[env]

      if (!projectId) {
        throw new EnvDoesntExistError(env)
      }

      return projectId
    }

    if (this.default && !skipDefault) {
      return this.default
    }

    return null
  }

  public getEnvironmentName(projectId: string): {envName: string} | null {
    const envName = Object.keys(this.env.environments).find(key => {
      const projectEnv = this.env.environments[key]
      return projectEnv === projectId
    })

    if (envName) {
      return {
        envName,
      }
    }

    return null
  }

}
