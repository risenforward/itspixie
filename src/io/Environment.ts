import * as fs from 'fs'
import * as yaml from 'js-yaml'
import { EnvironmentConfig, Environments, ProjectEnvironment } from '../types'
import * as path from 'path'
import client from './Client'

const envPath = path.join(process.cwd(), '.graphcool.env')

class Environment {
  env: EnvironmentConfig

  private initEmptyEnvironment() {
    this.env = {
      default: null,
      environments: {}
    }
  }

  get default(): ProjectEnvironment | null {
    if (this.env.default) {
      return this.env.environments[this.env.default]
    }

    return null
  }

  public load() {
    if (fs.existsSync(envPath)) {
      this.env = yaml.safeLoad(fs.readFileSync(envPath, 'utf-8'))
    } else {
      this.initEmptyEnvironment()
    }
  }

  public save() {
    const file = yaml.safeDump(this.env)
    fs.writeFileSync(envPath, file)
  }

  public async set(name: string, projectId: string) {
    const version = await client.getProjectVersion(projectId)

    this.env.environments[name] = {
      projectId,
      version,
    }
  }

  public rename(oldName: string, newName: string) {

  }

  public remove(oldName: string, newName: string) {

  }

}

const env = new Environment()
env.load()

export default env
