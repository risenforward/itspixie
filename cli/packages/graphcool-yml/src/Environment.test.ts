import { Environment } from './Environment'
import { getTmpDir } from './test/getTmpDir'
import * as path from 'path'
import * as fs from 'fs-extra'

export function makeEnv(rc?: string) {
  const tmpDir = getTmpDir()
  return new Environment(tmpDir)
}

describe('Environment', () => {
  test('non-existent global graphcool rc', async () => {
    const env = makeEnv()
    await env.load({})
    expect(env.clusters).toMatchSnapshot()
  })
  test('empty global graphcool rc', async () => {
    const env = makeEnv('')
    await env.load({})
    expect(env.clusters).toMatchSnapshot()
  })
  test('sets the platform token correctly', async () => {
    const env = makeEnv(`platformToken: asdf`)
    await env.load({})
    expect(env.clusters).toMatchSnapshot()
  })
  test('interpolates env vars', async () => {
    process.env.SPECIAL_TEST_ENV_VAR = 'this-is-so-special'
    const env = makeEnv(`platformToken: \${env:SPECIAL_TEST_ENV_VAR}`)
    await env.load({})
    expect(env.clusters).toMatchSnapshot()
  })
  test('loads multiple cluster definitions correctly + gives cluster by name', async () => {
    const rc = `clusters:
    local:
      host: 'http://localhost:60000'
    remote:
      host: 'https://remote.graph.cool'
      clusterSecret: 'here-is-a-token'
  `
    const env = makeEnv(rc)
    await env.load({})
    expect(env.clusters).toMatchSnapshot()

    const cluster = env.clusterByName('remote')
    expect(cluster).toMatchSnapshot()
  })
})
