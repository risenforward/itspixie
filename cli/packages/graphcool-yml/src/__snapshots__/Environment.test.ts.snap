// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environment empty global graphcool rc 1`] = `
Array [
  Cluster {
    "baseUrl": "https://database-beta.graph.cool",
    "clusterToken": undefined,
    "local": false,
    "name": "shared-public-demo",
  },
]
`;

exports[`Environment interpolates env vars 1`] = `
Array [
  Cluster {
    "baseUrl": "https://database-beta.graph.cool",
    "clusterToken": undefined,
    "local": false,
    "name": "shared-public-demo",
  },
]
`;

exports[`Environment loads multiple cluster definitions correctly + gives cluster by name 1`] = `
Array [
  Cluster {
    "baseUrl": "https://database-beta.graph.cool",
    "clusterToken": undefined,
    "local": false,
    "name": "shared-public-demo",
  },
]
`;

exports[`Environment loads multiple cluster definitions correctly + gives cluster by name 2`] = `undefined`;

exports[`Environment non-existent global graphcool rc 1`] = `
Array [
  Cluster {
    "baseUrl": "https://database-beta.graph.cool",
    "clusterToken": undefined,
    "local": false,
    "name": "shared-public-demo",
  },
]
`;

exports[`Environment persists .graphcoolrc correctly 1`] = `
"graphcool-1.0:
  clusters:
    cluster:
      host: 'http://localhost:60000'
      clusterSecret: ''
"
`;

exports[`Environment persists .graphcoolrc correctly 2`] = `
Array [
  Cluster {
    "baseUrl": "https://database-beta.graph.cool",
    "clusterToken": undefined,
    "local": false,
    "name": "shared-public-demo",
  },
  Cluster {
    "baseUrl": "http://localhost:60000",
    "clusterToken": "",
    "local": true,
    "name": "cluster",
  },
]
`;

exports[`Environment sets the platform token correctly 1`] = `
Array [
  Cluster {
    "baseUrl": "https://database-beta.graph.cool",
    "clusterToken": undefined,
    "local": false,
    "name": "shared-public-demo",
  },
]
`;
